modular_skyrat/modules/assault_operatives/code/interrogator.dm

```diff
@@ -0,0 +1,215 @@
+		balloon_alert_to_viewers("door open!")
+		return
+	if(!is_station_level(z))
+		balloon_alert_to_viewers("no comms link!")
+		return
+	if(!ishuman(occupant))
+		balloon_alert_to_viewers("invalid target dna!")
+		return
+	human_occupant = occupant
+	if(human_occupant.stat == DEAD)
+		balloon_alert_to_viewers("occupant is dead!")
+		return
+	var/datum/objective/interrogate/objective = get_goldeneye_target(human_occupant)
+	if(!objective || objective.goldeneye_key_uploaded) // Preventing abuse by method of duplication.
+		balloon_alert_to_viewers("no goldeneye data!")
+		playsound(src, 'sound/machines/scanbuzz.ogg', 100)
+		return
+
+	start_extract()
+
+/obj/machinery/interrogator/proc/start_extract()
+	to_chat(human_occupant, span_userdanger("You feel dread wash over you as you hear the door on [src] lock!"))
+	locked = TRUE
+	processing = TRUE
+	say("Starting DNA data extraction!")
+	timer_id = addtimer(CALLBACK(src, .proc/stage_one), rand(STAGE_PROCESS_TIME_LOWER, STAGE_PROCESS_TIME_UPPER), TIMER_STOPPABLE|TIMER_UNIQUE) //Random times so crew can't anticipate exactly when it will drop.
+	update_appearance()
+
+/obj/machinery/interrogator/proc/stage_one()
+	if(!check_requirements())
+		say("Critical error! Aborting.")
+		playsound(src, 'sound/machines/scanbuzz.ogg', 100)
+		return
+	to_chat(human_occupant, span_danger("As [src] whirrs to life you feel some cold metal restraints deploy around you, you can't move!"))
+	playsound(loc, 'sound/items/rped.ogg', 60)
+	say("Stage one complete!")
+	minor_announce("SECURITY BREACH DETECTED, NETWORK COMPROMISED! READING COORDINATES...", "GoldenEye Defence Network")
+	timer_id = addtimer(CALLBACK(src, .proc/stage_two), rand(STAGE_PROCESS_TIME_LOWER, STAGE_PROCESS_TIME_UPPER), TIMER_STOPPABLE|TIMER_UNIQUE)
+
+/obj/machinery/interrogator/proc/stage_two()
+	if(!check_requirements())
+		say("Critical error! Aborting.")
+		playsound(src, 'sound/machines/scanbuzz.ogg', 100)
+		return
+	to_chat(human_occupant, span_userdanger("You feel a sharp pain as a drill penetrates your skull, it's unbearable!"))
+	human_occupant.emote("scream")
+	human_occupant.apply_damage(30, BRUTE, BODY_ZONE_HEAD)
+	playsound(src, 'sound/effects/wounds/blood1.ogg', 100)
+	playsound(src, 'sound/items/drill_use.ogg', 100)
+	say("Stage two complete!")
+	minor_announce("SECURITY BREACH DETECTED, NETWORK COMPROMISED! COORDINATES: [x], [y], [z]", "GoldenEye Defence Network")
+	timer_id = addtimer(CALLBACK(src, .proc/stage_three), rand(STAGE_PROCESS_TIME_LOWER, STAGE_PROCESS_TIME_UPPER), TIMER_STOPPABLE|TIMER_UNIQUE)
+
+/obj/machinery/interrogator/proc/stage_three()
+	if(!check_requirements())
+		say("Critical error! Aborting.")
+		playsound(src, 'sound/machines/scanbuzz.ogg', 100)
+		return
+	to_chat(human_occupant, span_userdanger("You feel something penetrating your brain, it feels as though your childhood memories are fading! Please, make it stop! After a moment of slience you realise you can't remember what happened to you!"))
```

Comment:
may want to add some message a la the revival "you don't remember events to your death" message if the intent is to make them forget

Reply:
It doesn't kill you though.